Loop/Managers/LoopDataManager.swift

```diff
@@ -187,21 +189,28 @@ final class LoopDataManager: LoopSettingsAlerterDelegate {
     }
 
     /// Loop-related settings
-    ///
-    /// These are not thread-safe.
 
-    @Published var settings: LoopSettings {
-        didSet {
-            guard settings != oldValue else {
+    private var lockedSettings: Locked<LoopSettings>
+
+    var settings: LoopSettings {
+        get {
+            lockedSettings.value
+        }
+        set {
```

Comment:
So in other places where we've wrapped a struct in `Locked`, we've usually not provided a setter like this. 

Consider this code:
```
settings.glucoseTargetRangeSchedule = newValue.glucoseTargetRangeSchedule
```

On the executing thread, this ends up being separate operations:
1) call the getter for `settings` to get the value of settings, which locks, reads, and unlocks `lockedSettings`.
2) call the setter for `settings` with an a new `LoopSettings` object (the struct returned from step 1 with an updated glucoseTargetRangeSchedule), which locks writes, and unlocks `lockedSettings`.

Other threads could execute code between 1 and 2, writing to settings, and those writes would be clobbered.

So that's why you see us having funcs like `mutateState` or `setStateWithResult`, that can wrap a block of code that reads/writes the struct.